### PR TITLE
TS5: Pre-release 'build-tools' to pick up incremental build fix

### PR DESCRIPTION
**Pull request by DLehenbauer**
This pre-release version of 'build-tools' to pick up @curtisman's change to invokes the package's local TypeScript when computing the file hashes (supports incremental builds with mixed versions of TypeScript!)

Also picks up @CraigMacomber's change to 'fluid-type-test-generator' that allows more permissive comparisons between nominally typed / branded string and numbers.  This is to avoid identical 'const enum' declarations being reported as unassignable with each other.

This removed most/all breaking changes currently reported for these files:
packages/common/driver-definitions/package.json
packages/drivers/odsp-driver-definitions/package.json
packages/loader/driver-utils/package.json
packages/utils/odsp-doclib-utils/package.json

### CODE REVIEW

File: azure/packages/azure-client/package.json [hunk 1]
```diff
@@ -62,7 +62,7 @@
 		"@fluidframework/azure-client-previous": "npm:@fluidframework/azure-client@2.0.0-internal.6.3.0",
```
Comment (DLehenbauer) [inline azure/packages/azure-client/package.json:62]: This PR only contains three kinds of changes:

1.  Bump build-tools to prerelease version "0.24.0-192663".
2.  Regenerated type test files.  (*.generated.ts)
3.  Removing type tests that now pass from 'package.json' and *.generated.ts.

All changes were performed by tools.  I'll "pre-review" the first two.

File: azure/packages/azure-client/package.json [hunk 2]
```diff
@@ -62,7 +62,7 @@
...
 		"@fluidframework/azure-local-service": "workspace:~",
 		"@fluidframework/build-common": "^2.0.0",
-		"@fluidframework/build-tools": "^0.22.0",
+		"@fluidframework/build-tools": "0.24.0-192663",
```
Comment (DLehenbauer) [inline azure/packages/azure-client/package.json:65]: Example of #1:  Bumping 'build-tools'.  Okay to skim past these.

File: azure/packages/azure-client/src/test/types/validateAzureClientPrevious.generated.ts [hunk 3]
```diff
@@ -6,12 +6,20 @@
...
+// See 'build-tools/src/type-test-generator/compatibility.ts' for more information.
+type TypeOnly<T> = T extends number
+	? number
+	: T extends string
+	? string
+	: T extends boolean | bigint | symbol
+	? T
+	: {
+			[P in keyof T]: TypeOnly<T[P]>;
+	  };
```
Comment (DLehenbauer) [inline azure/packages/azure-client/src/test/types/validateAzureClientPrevious.generated.ts:22]: Example of 2: Regenerating type test to update boiler plate.  Skim past these as well.

File: packages/common/driver-definitions/package.json [hunk 4]
```diff
@@ -63,35 +63,8 @@
...
-			},
-			"InterfaceDeclaration_IThrottlingWarning": {
-				"backCompat": false
-			},
 			"VariableDeclaration_DriverErrorTypes": {
 				"forwardCompat": false
-			},
-			"TypeAliasDeclaration_DriverErrorTypes": {
-				"backCompat": false
 			}
```
Comment (DLehenbauer) [inline packages/common/driver-definitions/package.json:68]: First instance of 3: Type tests that now pass due to the more permissive comparison of string-like and number-like.

In this case, the new permissive TypeOnly check is allowing an enum-like intersection of string literals (i.e., 'genericError' | 'networkError' | ...) as just a 'string'.

File: packages/drivers/odsp-driver-definitions/package.json [hunk 5]
```diff
@@ -53,16 +53,6 @@
...
-				"backCompat": false
-			},
-			"InterfaceDeclaration_IOdspError": {
-				"backCompat": false
-			},
-			"TypeAliasDeclaration_OdspError": {
-				"backCompat": false
-			}
-		}
+		"broken": {}
```
Comment (DLehenbauer) [inline packages/drivers/odsp-driver-definitions/package.json:56]: Next instance of 3: Additional values added to 'OdspErrorType' enum no longer flagged as backCompat issue.

File: packages/loader/driver-utils/package.json [hunk 6]
```diff
@@ -98,28 +98,6 @@
...
-				"backCompat": false
-			},
-			"ClassDeclaration_UsageError": {
-				"backCompat": false
-			},
-			"ClassDeclaration_ThrottlingError": {
-				"backCompat": false
-			}
-		}
+		"broken": {}
```
Comment (DLehenbauer) [inline packages/loader/driver-utils/package.json:101]: const strings from FluidErrorType not comparing as equal.

File: packages/utils/odsp-doclib-utils/package.json [hunk 7]
```diff
@@ -93,10 +93,6 @@
 		"typescript": "~4.5.5"
 	},
 	"typeValidation": {
-		"broken": {
-			"ClassDeclaration_OdspRedirectError": {
-				"backCompat": false
-			}
-		}
+		"broken": {}
```
Comment (DLehenbauer) [inline packages/utils/odsp-doclib-utils/package.json:96]: Change kind 3:  string constant 'errorType: DriverErrorType.fileNotFoundOrAccessDeniedError'.

File: packages/common/driver-definitions/package.json [hunk 8]
```diff
@@ -63,35 +63,8 @@
 	},
 	"typeValidation": {
 		"broken": {
-			"TypeAliasDeclaration_DriverError": {
-				"backCompat": false
-			},
-			"EnumDeclaration_DriverErrorType": {
```
Comment (DLehenbauer) [inline packages/common/driver-definitions/package.json:69]: Change kind 3: Type tests that now pass due to the more permissive comparison of string-like and number-like types.  The motivation is that TypeScript considers identical string/number literals as being unassignable to each other when coming from different (but identical) enum declarations, creating more noise than value in our backCompat tests.

In this case, the remaining forwardCompat breaking change remains, due to adding new values to the DriverErrorTypes enum-like, but types that are descriminated by 'DriverErrorTypes.someConstant' are no longer flagged (and shouldn't be, as the string value of 'someConstant' is unchanged.)

**Review by CraigMacomber — APPROVED** [review]

**Comment by CraigMacomber** [comment]
Mixed TS version support will be great for allowing leaf packages like tree2 to be TS early adopters if needed, and all those reductions to incorrect broken API reports is excellent!

**Pull request merged**